zulipterminal/ui_tools/views.py

```diff
@@ -311,9 +311,34 @@ def __init__(self, topics_btn_list: List[Any], view: Any,
                  stream_button: Any) -> None:
         self.view = view
         self.log = urwid.SimpleFocusListWalker(topics_btn_list)
+        self.topics_btn_list = topics_btn_list
         self.stream_button = stream_button
         self.list_box = urwid.ListBox(self.log)
-        super(TopicsView, self).__init__(self.list_box)
+        self.topic_search_box = PanelSearchBox(self, 'SEARCH_TOPICS')
+        urwid.connect_signal(self.topic_search_box,
+                             'change', self.update_topics)
+        super(TopicsView, self).__init__(self.list_box, header=urwid.LineBox(
+            self.topic_search_box, tlcorner=u'─', tline=u'', lline=u'',
+            trcorner=u'─', blcorner=u'─', rline=u'',
+            bline=u'─', brcorner=u'─'
+        ))
+        self.search_lock = threading.Lock()
+
+    @asynch
+    def update_topics(self, search_box: Any, new_text: str) -> None:
+        if not self.view.controller.editor_mode:
+            return
+        # wait for any previously started search to finish to avoid
+        # displaying wrong topics list.
+        with self.search_lock:
+            topics_to_display = self.topics_btn_list.copy()
+            for topic_button in self.topics_btn_list:
+                if not topic_button.topic_name.lower().startswith(
+                        new_text.lower()):
+                    topics_to_display.remove(topic_button)
```

Comment:
You've moved to the locking version here; what do you feel about the list comprehension approach?

On a related note, this only matches the start of topics; we could (later?) support anywhere in the word, or similar?
Eg. "Install error" wouldn't be matched with "error".

Reply:
Seems fair to support the List comprehensions :+1: 

Agreed. Something along the lines of `match_user` in helper.py can be used for that.